### PR TITLE
Glue streaming job update

### LINKED ISSUE #16797: Glue job timeout can be 0, but TF validation requires min value of 1
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_glue_job

### Terraform Configuration Files

`
resource "aws_glue_job" "glue_job" {
  name = "glue_job"
  role_arn = iam_role_arn
  connections = []

  command {
    name = "gluestreaming"
    script_location = "s3://path.py"
    python_version = "3"
  }

  glue_version  = "1.0"
  max_retries = 0
  max_capacity = 1
 timeout = 0
}
`

```hcl

### Debug Output

`
Error: error updating Glue Job (glue_job): InvalidParameter: 1 validation error(s) found.
- minimum field value of 1, UpdateJobInput.JobUpdate.Timeout.
- `

### Expected Behavior

This glue job is defined as a streaming job, so a value of 0 means infinite timeout, which is valid. You can confirm this by creating a glue job using AWS UI and if you don't set a timeout, glue sets the state to timeout = 0. 

### Actual Behavior

Throw's validation error if timeout is set to 0.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

For streaming glue jobs, timeout = 0 is valid, for non-streaming jobs, the field is optional and there IS a default value and the one the provider sets currently, is correct.

### References

* #0000

### PR DESCRIPTION
**Pull request by gerrardcowburn**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #16797

Output from acceptance testing:

```
$make testacc TESTS=TestAccGlueJob PKG=glue
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/glue/... -v -count 1 -parallel 20 -run='TestAccGlueJob'  -timeout 180m
=== RUN   TestAccGlueJob_basic
=== PAUSE TestAccGlueJob_basic
=== RUN   TestAccGlueJob_basicStreaming
=== PAUSE TestAccGlueJob_basicStreaming
=== RUN   TestAccGlueJob_command
=== PAUSE TestAccGlueJob_command
=== RUN   TestAccGlueJob_defaultArguments
=== PAUSE TestAccGlueJob_defaultArguments
=== RUN   TestAccGlueJob_nonOverridableArguments
=== PAUSE TestAccGlueJob_nonOverridableArguments
=== RUN   TestAccGlueJob_description
=== PAUSE TestAccGlueJob_description
=== RUN   TestAccGlueJob_glueVersion
=== PAUSE TestAccGlueJob_glueVersion
=== RUN   TestAccGlueJob_executionProperty
=== PAUSE TestAccGlueJob_executionProperty
=== RUN   TestAccGlueJob_maxRetries
=== PAUSE TestAccGlueJob_maxRetries
=== RUN   TestAccGlueJob_notificationProperty
=== PAUSE TestAccGlueJob_notificationProperty
=== RUN   TestAccGlueJob_tags
=== PAUSE TestAccGlueJob_tags
=== RUN   TestAccGlueJob_streamingTimeout
=== PAUSE TestAccGlueJob_streamingTimeout
=== RUN   TestAccGlueJob_timeout
=== PAUSE TestAccGlueJob_timeout
=== RUN   TestAccGlueJob_security
=== PAUSE TestAccGlueJob_security
=== RUN   TestAccGlueJob_workerType
=== PAUSE TestAccGlueJob_workerType
=== RUN   TestAccGlueJob_pythonShell
=== PAUSE TestAccGlueJob_pythonShell
=== RUN   TestAccGlueJob_maxCapacity
=== PAUSE TestAccGlueJob_maxCapacity
=== RUN   TestAccGlueJob_disappears
=== PAUSE TestAccGlueJob_disappears
=== CONT  TestAccGlueJob_basic
=== CONT  TestAccGlueJob_notificationProperty
=== CONT  TestAccGlueJob_executionProperty
=== CONT  TestAccGlueJob_timeout
=== CONT  TestAccGlueJob_glueVersion
=== CONT  TestAccGlueJob_streamingTimeout
=== CONT  TestAccGlueJob_description
=== CONT  TestAccGlueJob_security
=== CONT  TestAccGlueJob_tags
=== CONT  TestAccGlueJob_nonOverridableArguments
=== CONT  TestAccGlueJob_maxRetries
=== CONT  TestAccGlueJob_disappears
=== CONT  TestAccGlueJob_pythonShell
=== CONT  TestAccGlueJob_command
=== CONT  TestAccGlueJob_maxCapacity
=== CONT  TestAccGlueJob_workerType
=== CONT  TestAccGlueJob_basicStreaming
=== CONT  TestAccGlueJob_defaultArguments
--- PASS: TestAccGlueJob_basic (128.36s)
--- PASS: TestAccGlueJob_disappears (138.74s)
--- PASS: TestAccGlueJob_basicStreaming (149.65s)
--- PASS: TestAccGlueJob_description (235.26s)
--- PASS: TestAccGlueJob_defaultArguments (235.66s)
--- PASS: TestAccGlueJob_streamingTimeout (236.81s)
--- PASS: TestAccGlueJob_timeout (237.15s)
--- PASS: TestAccGlueJob_notificationProperty (244.28s)
--- PASS: TestAccGlueJob_maxRetries (245.41s)
--- PASS: TestAccGlueJob_maxCapacity (246.62s)
--- PASS: TestAccGlueJob_executionProperty (246.87s)
--- PASS: TestAccGlueJob_security (249.15s)
--- PASS: TestAccGlueJob_command (255.52s)
--- PASS: TestAccGlueJob_nonOverridableArguments (256.43s)
--- PASS: TestAccGlueJob_pythonShell (322.16s)
--- PASS: TestAccGlueJob_tags (329.92s)
--- PASS: TestAccGlueJob_workerType (334.76s)
--- PASS: TestAccGlueJob_glueVersion (341.94s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/glue       347.402s
```

CHANGELOG.md
```
* resource/aws_glue_job: Add support for `gluestreaming` Command Name by removing `Timeout` as a default parameter on Jobs, moving to optional, and adding acc tests ([#16797]https://github.com/hashicorp/terraform-provider-aws/issues/16797)
```

### CODE REVIEW

File: internal/service/glue/job.go [hunk 1]
```diff
@@ -132,9 +132,10 @@ func ResourceJob() *schema.Resource {
 			"tags":     tftags.TagsSchema(),
 			"tags_all": tftags.TagsSchemaComputed(),
 			"timeout": {
-				Type:     schema.TypeInt,
-				Optional: true,
-				Default:  2880,
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ValidateFunc: validation.IntAtLeast(1),
```
Comment (drewmullen) [inline internal/service/glue/job.go:138]: Does the API have an upper bounds?

**Review by drewmullen — CHANGES_REQUESTED** [review]
1 small Q and please create your [changelog entry](https://github.com/hashicorp/terraform-provider-aws/blob/main/docs/contributing/pullrequest-submission-and-lifecycle.md#changelog-process). looks good!

File: internal/service/glue/job.go (hunk 1 above)
Comment (gerrardcowburn) [reply internal/service/glue/job.go:138]: Not per the documentation, and testing high values such as 300000 work as expected

**Review by drewmullen — CHANGES_REQUESTED** [review]
Can you also update the documentation to include a reference to `gluestreaming` and a brief note about how timeout behaves differently:
- [here](https://github.com/hashicorp/terraform-provider-aws/blob/1af821c85349593a32ef15a84679e74103101bf3/website/docs/r/glue_job.html.markdown?plain=1#L92)
- and [here](https://github.com/hashicorp/terraform-provider-aws/blob/1af821c85349593a32ef15a84679e74103101bf3/website/docs/r/glue_job.html.markdown?plain=1#L85)

**Comment by gerrardcowburn** [comment]
Done! :)

**Review by drewmullen — APPROVED** [review]
LGTM!

**Review by DrFaust92 — APPROVED** [review]
LGTM

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTS=TestAccGlueJob_ PKG=glue ACCTEST_PARALLELISM=5
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/glue/... -v -count 1 -parallel 5 -run='TestAccGlueJob_'  -timeout 180m
=== RUN   TestAccGlueJob_basic
=== PAUSE TestAccGlueJob_basic
=== RUN   TestAccGlueJob_basicStreaming
=== PAUSE TestAccGlueJob_basicStreaming
=== RUN   TestAccGlueJob_command
=== PAUSE TestAccGlueJob_command
=== RUN   TestAccGlueJob_defaultArguments
=== PAUSE TestAccGlueJob_defaultArguments
=== RUN   TestAccGlueJob_nonOverridableArguments
=== PAUSE TestAccGlueJob_nonOverridableArguments
=== RUN   TestAccGlueJob_description
=== PAUSE TestAccGlueJob_description
=== RUN   TestAccGlueJob_glueVersion
=== PAUSE TestAccGlueJob_glueVersion
=== RUN   TestAccGlueJob_executionProperty
=== PAUSE TestAccGlueJob_executionProperty
=== RUN   TestAccGlueJob_maxRetries
=== PAUSE TestAccGlueJob_maxRetries
=== RUN   TestAccGlueJob_notificationProperty
=== PAUSE TestAccGlueJob_notificationProperty
=== RUN   TestAccGlueJob_tags
=== PAUSE TestAccGlueJob_tags
=== RUN   TestAccGlueJob_streamingTimeout
=== PAUSE TestAccGlueJob_streamingTimeout
=== RUN   TestAccGlueJob_timeout
=== PAUSE TestAccGlueJob_timeout
=== RUN   TestAccGlueJob_security
=== PAUSE TestAccGlueJob_security
=== RUN   TestAccGlueJob_workerType
=== PAUSE TestAccGlueJob_workerType
=== RUN   TestAccGlueJob_pythonShell
=== PAUSE TestAccGlueJob_pythonShell
=== RUN   TestAccGlueJob_maxCapacity
=== PAUSE TestAccGlueJob_maxCapacity
=== RUN   TestAccGlueJob_disappears
=== PAUSE TestAccGlueJob_disappears
=== CONT  TestAccGlueJob_basic
=== CONT  TestAccGlueJob_notificationProperty
=== CONT  TestAccGlueJob_description
=== CONT  TestAccGlueJob_defaultArguments
=== CONT  TestAccGlueJob_executionProperty
--- PASS: TestAccGlueJob_basic (131.42s)
=== CONT  TestAccGlueJob_glueVersion
--- PASS: TestAccGlueJob_description (211.79s)
=== CONT  TestAccGlueJob_workerType
--- PASS: TestAccGlueJob_defaultArguments (298.67s)
=== CONT  TestAccGlueJob_disappears
--- PASS: TestAccGlueJob_executionProperty (303.57s)
=== CONT  TestAccGlueJob_maxCapacity
--- PASS: TestAccGlueJob_disappears (68.17s)
=== CONT  TestAccGlueJob_pythonShell
--- PASS: TestAccGlueJob_notificationProperty (419.51s)
=== CONT  TestAccGlueJob_command
--- PASS: TestAccGlueJob_glueVersion (303.85s)
=== CONT  TestAccGlueJob_basicStreaming
--- PASS: TestAccGlueJob_workerType (261.75s)
=== CONT  TestAccGlueJob_nonOverridableArguments
--- PASS: TestAccGlueJob_basicStreaming (73.06s)
=== CONT  TestAccGlueJob_maxRetries
--- PASS: TestAccGlueJob_command (141.38s)
=== CONT  TestAccGlueJob_timeout
--- PASS: TestAccGlueJob_pythonShell (212.33s)
=== CONT  TestAccGlueJob_security
--- PASS: TestAccGlueJob_maxCapacity (278.03s)
=== CONT  TestAccGlueJob_tags
--- PASS: TestAccGlueJob_nonOverridableArguments (140.00s)
=== CONT  TestAccGlueJob_streamingTimeout
--- PASS: TestAccGlueJob_maxRetries (145.56s)
--- PASS: TestAccGlueJob_timeout (164.41s)
--- PASS: TestAccGlueJob_security (191.38s)
--- PASS: TestAccGlueJob_streamingTimeout (254.13s)
--- PASS: TestAccGlueJob_tags (289.21s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/glue	874.488s
```

**Pull request merged**